qualtran/bloqs/rotations/programmable_ancilla_rotation.py

```diff
@@ -0,0 +1,161 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from functools import cached_property
+from typing import Set
+
+import numpy as np
+import sympy
+from attrs import frozen
+
+from qualtran import Bloq, bloq_example, BloqBuilder, QBit, Register, Side, Signature, SoquetT
+from qualtran.bloqs.basic_gates import CNOT, Hadamard, Rz, XGate
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+from qualtran.symbolics import ceil, is_symbolic, log2, SymbolicFloat, SymbolicInt
+
+
+@frozen
+class RzResourceState(Bloq):
+    r"""Resource qubit with state $Rz(\phi) |+\rangle$"""
```

Comment:
```suggestion
    r"""Resource qubit prepared in state $Rz(\phi) |+\rangle$ = $|0\rangle + e^{i\phi}|1\rangle$"""
```

Although the reference says the prepared state is $|0\rangle + e^{i\phi}|1\rangle$, if you take the definition of $Rz(t) = e^{-i Z t /2}$ as commonly defined, the prepared state would be $e^{-i\phi/2}|0\rangle + e^{i\phi/2}|1\rangle$.  It's worth verifying that the procedure works as expected with the right global phase.

Reply:
fixed: used ZPow instead of Rz